src/btree/bt_sync.c

```diff
@@ -388,6 +388,12 @@ __sync_page_skip(WT_SESSION_IMPL *session, WT_REF *ref, void *context, bool *ski
     if (ref->state != WT_REF_DISK)
         return (0);
 
+    /* Don't read pages into cache during startup or shutdown phase. */
+    if (F_ISSET(S2C(session), WT_CONN_RECOVERING | WT_CONN_CLOSING_TIMESTAMP)) {
+        *skipp = true;
```

Comment:
Can you explain why this was necessary when excluding the cleanup work at shutdown time?

Reply:
Earlier checkpoint used to traverse only the pages in the cache but with WT-6058, we read all the pages into the cache if they have obsolete records to be aborted. But the flow of avoiding the cleanup and reading the pages into the cache is different, so the check must be repeated here also to avoid reading the unnecessary pages into cache if we don't need them.

Reply:
I see now that you explained in a comment. It makes sense.